clippy_lints/src/utils/internal_lints.rs

```diff
@@ -395,3 +401,111 @@ fn is_trigger_fn(fn_kind: FnKind<'_>) -> bool {
+                                    span_snippet,
+                                    msg_snippet,
+                                    help_snippet,
+                                    sugg_snippet,
+                                    applicability_snippet
+                                ),
+                                Applicability::MachineApplicable
+                            );
+                        }
+                    },
+                    "span_help" => {
+                        let span_snippet_of_span_call = snippet(cx, span_call_args[0].span);
+                        let help_snippet = snippet(cx, span_call_args[1].span);
+
+                        if span_snippet == span_snippet_of_span_call {
+                            span_lint_and_sugg(
+                                cx,
+                                COLLAPSIBLE_SPAN_LINT_CALLS,
+                                expr.span,
+                                "this call is collapsible",
+                                "collapse into",
+                                format!(
+                                    "span_lint_and_help({}, {}, {}, {},{})",
+                                    cx_snippet,
+                                    lint_snippet,
+                                    span_snippet,
+                                    msg_snippet,
+                                    help_snippet
+                                ),
+                                Applicability::MachineApplicable
+                            );
+                        }
+                    },
+                    "span_note" => {
+                        let span_snippet_of_span_call = snippet(cx, span_call_args[0].span);
+                        let note_snippet = snippet(cx, span_call_args[1].span);
+
+                        if span_snippet == span_snippet_of_span_call {
+                            span_lint_and_sugg(
+                                cx,
+                                COLLAPSIBLE_SPAN_LINT_CALLS,
+                                expr.span,
+                                "this call is collspible",
+                                "collapse into",
+                                format!(
+                                    "span_lint_and_note({},{}, {}, {}, {})",
+                                    cx_snippet,
+                                    lint_snippet,
+                                    span_snippet,
+                                    msg_snippet,
+                                    note_snippet
+                                ),
+                                Applicability::MachineApplicable
+                            );
+                        }
+                    },
+                    _  => (),
+                }
+            }
+        }
+    }
+}
+
+fn snippet(cx: &EarlyContext<'_>, span: Span) -> String {
+    snippet_opt(cx, span).expect("should be able to retrive span.")
+}
```

Comment:
Please use `snippet_with_applicability`, or just `snippet`, since it is an internal lint and we don't have to be too careful with the applicability.